### PR TITLE
add filter command

### LINKED ISSUE #51: As a HR manager, I can filter the list by Department and Tags

### PR DESCRIPTION
**Pull request by laybach**

### CODE REVIEW

**Review by howenyap — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.PersonContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
```
Comment (ziyi22) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:11]: Some typo in the java doc comments. Should be 'whose department or tags'

File: src/test/java/seedu/address/logic/commands/FilterCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    @Test
+    public void execute_multipleKeywords_multiplePersonsFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        PersonContainsKeywordsPredicate predicate = preparePredicate("Marketing Production");
+        FilterCommand command = new FilterCommand(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(CARL, DANIEL), model.getFilteredPersonList());
```
Comment (ziyi22) [inline src/test/java/seedu/address/logic/commands/FilterCommandTest.java:42]: Great work on testing the scenario of filtering based on multiple keywords.

File: src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
```
Comment (ziyi22) [inline src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java:9]: Javadoc typos

**Pull request merged**